xds/internal/client/rds.go

```diff
@@ -126,7 +127,8 @@ func (v2c *v2Client) handleRDSResponse(resp *xdspb.DiscoveryResponse) error {
 // field must be set.  Inside that route message, the cluster field will
 // contain the clusterName we are looking for.
 func getClusterFromRouteConfiguration(rc *xdspb.RouteConfiguration, target string) string {
-	host, _, err := net.SplitHostPort(target)
+	// TODO: return error for better error logging and nack.
```

Comment:
Returning empty string from here already makes `handleRDSResponse` return an error. So, should we just log the error here and remove the TODO or is there more to be done?

Reply:
Explained more in the TODO. I was think we should populate error details for nack. But getting error details is tricky because it's a slice.